assertions/scanner.go

```diff
@@ -440,11 +444,32 @@ func (s *Manager) keepTryingAssertionConfirmation(ctx context.Context, assertion
 			}
 			err = s.chain.ConfirmAssertionByTime(ctx, assertionHash)
 			if err != nil {
+				if strings.Contains(err.Error(), protocol.BeforeDeadlineAssertionConfirmationError) {
+					creationInfo, err := s.chain.ReadAssertionCreationInfo(ctx, assertionHash)
+					if err != nil {
+						continue
+					}
+					prevCreationInfo, err := s.chain.ReadAssertionCreationInfo(ctx, protocol.AssertionHash{Hash: creationInfo.ParentAssertionHash})
+					if err != nil {
+						continue
+					}
+					latestHeader, err := s.chain.Backend().HeaderByNumber(ctx, nil)
+					if err != nil {
+						continue
+					}
+					timeLeft := (creationInfo.CreationBlock + prevCreationInfo.ConfirmPeriodBlocks) - latestHeader.Number.Uint64()
+					if s.averageTimeForBlockCreation*time.Duration(timeLeft) > s.confirmationAttemptInterval {
```

Comment:
This condition is a bit gnarly, perhaps it can be refactored into its own function? Moreover, we don't need to tick each time if we are before this interval. Perhaps at the start of the function, we can do a `<-time.After(timeLeft)`, and then start ticking at confirmationAttemptInterval in a for select loop

Reply:
Done.